### ISSUE TITLE
azure_rm_storageblob - AttributeError: 'Cloud' object has no attribute 'decode'

### ISSUE THREAD

**Issue by filippoantoniolussana** [ansible/ansible_collections_azure#1337]
I'm getting the following error when trying to use azure_rm_storageblob. (Azure collection 2.0.0)
**Error:**
```
AttributeError: 'Cloud' object has no attribute 'decode'\n",
    "module_stdout": "",
    "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
    "rc": 1
}
```

**Related task:**
```
- name: Download the file locally
      azure.azcollection.azure_rm_storageblob:
        resource_group: 
        auth_source: cli
        subscription_id: 
        storage_account_name: 
        container: 
        blob: 
        dest: 
      delegate_to: localhost
```

**Ansible version:**
```
ansible [core 2.15.6]
  config file = None
  configured module search path = ['/home/filippo/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/filippo/.local/lib/python3.10/site-packages/ansible
  ansible collection location = /home/filippo/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.12 (main, Jun 11 2023, 05:26:28) [GCC 11.4.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

**Comment by Anthirian** [comment]
The same error occurs with `azure_rm_adapplication`:

```
TASK [Check for previously created Entra ID App Registrations] ***************************************************************
zondag 19 november 2023  17:03:56 +0100 (0:00:00.057)       0:00:00.141 *******
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: AttributeError: 'Cloud' object has no attribute 'decode'
fatal: [localhost]: FAILED! => changed=false
  module_stderr: |-
    Traceback (most recent call last):
      File "/home/geert/.ansible/tmp/ansible-tmp-1700409836.1228435-29176-216044148215724/AnsiballZ_azure_rm_adapplication_info.py", line 107, in <module>
        _ansiballz_main()
      File "/home/geert/.ansible/tmp/ansible-tmp-1700409836.1228435-29176-216044148215724/AnsiballZ_azure_rm_adapplication_info.py", line 99, in _ansiballz_main
        invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)
      File "/home/geert/.ansible/tmp/ansible-tmp-1700409836.1228435-29176-216044148215724/AnsiballZ_azure_rm_adapplication_info.py", line 47, in invoke_module
        runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_adapplication_info', init_globals=dict(_module_fqn='ansible_collections.azure.azcollection.plugins.modules.azure_rm_adapplication_info', _modlib_path=modl
ib_path),
      File "/usr/lib64/python3.10/runpy.py", line 224, in run_module
        return _run_module_code(code, init_globals, run_name, mod_spec)
      File "/usr/lib64/python3.10/runpy.py", line 96, in _run_module_code
        _run_code(code, mod_globals, init_globals,
      File "/usr/lib64/python3.10/runpy.py", line 86, in _run_code
        exec(code, run_globals)
      File "/tmp/ansible_azure.azcollection.azure_rm_adapplication_info_payload_qp01jamp/ansible_azure.azcollection.azure_rm_adapplication_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_adapplication_info.py", line 192,
 in <module>
      File "/tmp/ansible_azure.azcollection.azure_rm_adapplication_info_payload_qp01jamp/ansible_azure.azcollection.azure_rm_adapplication_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_adapplication_info.py", line 188,
 in main
      File "/tmp/ansible_azure.azcollection.azure_rm_adapplication_info_payload_qp01jamp/ansible_azure.azcollection.azure_rm_adapplication_info_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_adapplication_info.py", line 124,
 in __init__
      File "/tmp/ansible_azure.azcollection.azure_rm_adapplication_info_payload_qp01jamp/ansible_azure.azcollection.azure_rm_adapplication_info_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 439, in __i
nit__
      File "/tmp/ansible_azure.azcollection.azure_rm_adapplication_info_payload_qp01jamp/ansible_azure.azcollection.azure_rm_adapplication_info_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py", line 1440, in __
init__
      File "/usr/lib64/python3.10/urllib/parse.py", line 399, in urlparse
        url, scheme, _coerce_result = _coerce_args(url, scheme)
      File "/usr/lib64/python3.10/urllib/parse.py", line 136, in _coerce_args
        return _decode_args(args) + (_encode_result,)
      File "/usr/lib64/python3.10/urllib/parse.py", line 120, in _decode_args
        return tuple(x.decode(encoding, errors) if x else '' for x in args)
      File "/usr/lib64/python3.10/urllib/parse.py", line 120, in <genexpr>
        return tuple(x.decode(encoding, errors) if x else '' for x in args)
    AttributeError: 'Cloud' object has no attribute 'decode'
  module_stdout: ''
  msg: |-
    MODULE FAILURE
    See stdout/stderr for the exact error
  rc: 1
```

```
$ ansible-galaxy collection list

# /home/geert/.ansible/collections/ansible_collections
Collection                    Version
----------------------------- -------
azure.azcollection            2.0.0
```

```
$ ansible-version
ansible [core 2.15.5]
  config file = /home/geert/tools/red-team/private-infrastructure.git/consent-phishing/ansible.cfg
  configured module search path = ['/home/geert/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/virtualenvs/ansible/lib/python3.10/site-packages/ansible
  ansible collection location = /home/geert/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/virtualenvs/ansible/bin/ansible
  python version = 3.10.13 (main, Aug 29 2023, 13:29:07) [GCC] (/opt/virtualenvs/ansible/bin/python)
  jinja version = 3.1.2
  libyaml = True
```

**Comment by Fred-sun** [comment]
fixes by #1340

**Comment by harshavmb** [comment]
Almost all the ARM resources are impacted with AZ-CLI auth option, not just `storage_account`, `azure_rm_adapplication`. 

As fix may take some more time, in the meantime which collection version could be used?
